dart/dynamics/BodyNode.cpp

```diff
@@ -1339,6 +1339,14 @@ void BodyNode::notifyTransformUpdate()
 {
   notifyVelocityUpdate(); // Global Velocity depends on the Global Transform
 
+  // Jacobian calculations are dependent on the parent's world transform, but
+  // not on the world transform of their own BodyNode, so they must be dirtied
+  // regardless of whether the world transform of this BodyNode is already dirty
```

Comment:
I'm now thinking we might need two update tracks:

``notifyJacobianUpdate()`` would only deal with updating the Jacobians, and it would cut off at whichever BodyNode already knows its Jacobian is dirty (if a parent BodyNode's Jacobian is dirty, then all its children's Jacobians must be dirty as well). ``notifyTransformUpdate()`` will remain the same, but the very first thing it will do is call its own ``notifyJacobianUpdate()``.

However, ``notifyJacobianUpdate()`` will need to be a member of the ``JacobianNode`` class, not the BodyNode class. That way all ``JacobianNode`` types (such as ``EndEffector``) will make use of it.